### PR TITLE
Add a FLUTTER_MODE variable to the generated Xcconfig and fix snapshot generation on all targets.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Get rid of the DART_EXPERIMENTAL_INTERPRETER_FLAG.

cc @abarth

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Updated to fix debug targets as well. Another look?

**Comment by chinmaygarde** [comment]
Fixes https://github.com/flutter/flutter/issues/4357

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 1]
```diff
@@ -58,16 +58,17 @@ BuildApp() {
...
 
   if [[ $CURRENT_ARCH != "x86_64" ]]; then
-    local interpreter_flag="--interpreter"
-    if [[ "$DART_EXPERIMENTAL_INTERPRETER" != "1" ]]; then
-      interpreter_flag=""
+    local aot_flags=""
+    if [[ "$FLUTTER_MODE" == "debug" ]]; then
+      aot_flags="--interpreter --debug"
+    else
+      aot_flags="--${FLUTTER_MODE}"
```
Comment (abarth) [inline packages/flutter_tools/bin/xcode_backend.sh]: Can you set a default for `FLUTTER_MODE` (presumably release) and then use the mode to set the default value of framework_path?  Also, please update the hello_services docs.

**Comment by abarth** [comment]
LGTM w/ minor comment above.

**Comment by chinmaygarde** [comment]
Done. Rebased.

**Pull request merged**